app/jobs/application_submission_job.rb

```diff
@@ -13,17 +13,56 @@ def perform(user:)
     user.applications.includes(:lead_provider, :course).where(ecf_id: nil).each do |application|
       Services::Ecf::NpqProfileCreator.new(application:).call
 
+      tsf_funding_amount = if application.targeted_delivery_funding_eligibility
+                             application.tsf_primary_plus_eligibility ? 800 : 200
+                           else
+                             0
+                           end
```

Comment:
It feels odd to have this calculation occurring primarily as a point related to mailing shouldn't this be something on the application itself that can just be called directly?

Reply:
Not sure i fully follow what you mean? As in right at the end? i guess this could be brought forward a little